### DISCUSSION TITLE
PrivateVPN port forward API returned a bad response

### DISCUSSION THREAD

**Discussion by Omegatcu** [haugene/vpn-configs-contrib#88] [Q&A] [answered]
I run the docker on a NAS - a couple of weeks ago I started to have problems with seeding. 
I couldn't look into this before, but now have the time, and it seems there is something wrong with the port forwarding script.

I get the following error when the container starts:

> Port Forwarding (UPnP) Port forwarding successful! (upnp.c:307)
> Port Forwarding State changed from "Not forwarded" to "Forwarded" (port-forwarding.c:90)
> /scripts/transmission-post-start.sh returned 6 [This is a script with a cookie based cURL call]
> Transmission startup script complete.
> PrivateVPN port forward API returned a bad response
> curl encountered an error looking up forwarded port: 6

My Config:
> CREATE_TUN_DEVICE=true
> DROP_DEFAULT_ROUTE=
> ENABLE_UFW=false
> GLOBAL_APPLY_PERMISSIONS=true
> HEALTH_CHECK_HOST=google.com
> LOG_TO_STDOUT=true
> OPENVPN_OPTS=--mute-replay-warnings
> OPENVPN_PASSWORD=password
> OPENVPN_PROVIDER=PRIVATEVPN
> OPENVPN_USERNAME=username
> PEER_DNS=true
> PEER_DNS_PIN_ROUTES=true
> SELFHEAL=false
> UFW_ALLOW_GW_NET=false
> UFW_DISABLE_IPTABLES_REJECT=false
> UFW_EXTRA_PORTS=61666
> WEBPROXY_ENABLED=false
> WEBPROXY_PASSWORD=
> WEBPROXY_PORT=8118
> WEBPROXY_USERNAME=

Does somebody know what is going wrong? - I think maybe there is something wrong with calling cURL, but when I use it in the console it works.
Any Ideas what to do?

**Comment by pkishino** [comment]
Hmm, I’m using PrivateVPN as well and don’t see any problems here. Have you tried different servers? 
I’ll try and take a look later.
Soo you tried to run the script inside the container and it was ok?

**Reply by Omegatcu** [reply]
I did try different Servers that should have port forwarding. The container did run with port forwarding in the past until around the 3rd week in January.
I didn't try to run the script from the config in the container - I only see the message from the startup execution.
Can you tell me how I can start the script after the container is fully loaded, or how to debug the error?
I can read and edit code if necessary.

**Reply by pkishino** [reply]
You can run it manually inside the container, create a shell into the container :)

**Reply by Omegatcu** [reply]
Hello,
okay I used /etc/openvpn/privatevpn/update-port.sh and now it works.
Could it be that the script runs too early? I have another script that runs after transmission start that also seems to have problems until the container is fully loaded.

**Reply by Omegatcu** [reply]
So I restarted the container and on startup the scripts are not running - it seems cURL is not really working. After transmission is reading all the torrent files, it works when I manually use the script.
Sadly, the last hook for the startup is transmission-post-start.sh and that seems still to early.

**Comment by Omegatcu** [comment, marked as answer]
The Answer was to run the script again after the Container was up and running
Script Path: /etc/openvpn/privatevpn/update-port.sh

Thanks @pkishino

**Comment by pkishino** [comment]
Hmm, do you have any other scripts you run? I’ve restarted my container
multiple times and the port script runs just fine here..

**Reply by Omegatcu** [reply]
I'm so sorry, I didn't see this comment in my feed!
I have one script that posts my port to a network but that is executed after the connection is established and is just a cURL call in the post script.
I now have a cronjob that is triggered outside the container and that runs the update-port.sh as well as the cURL call in a regular interval.
